src/plugins/chart_expressions/expression_xy/public/components/xy_chart.tsx

```diff
@@ -1038,3 +1041,26 @@ export function XYChart({
     </div>
   );
 }
+
+const defaultLegendTitle = i18n.translate('expressionXY.xyChart.legendTitle', {
+  defaultMessage: 'Legend',
+});
+
+function getLegendTitle(
+  title: string | undefined,
+  layer?: CommonXYDataLayerConfig,
+  isTitleVisible?: boolean
+) {
+  if (!isTitleVisible) {
+    return undefined;
+  }
+  if (typeof title === 'string' && title.length > 0) {
+    return title;
+  }
+  if (!layer) {
+    return defaultLegendTitle;
+  }
+  return layer.splitAccessors?.[0]
+    ? getColumnByAccessor(layer.splitAccessors?.[0], layer?.table.columns)?.name
+    : defaultLegendTitle;
```

Comment:
Could simplify this if you want

```suggestion
  return layer?.splitAccessors?.[0]
    ? getColumnByAccessor(layer.splitAccessors?.[0], layer?.table.columns)?.name
    : defaultLegendTitle;
```